diff -Nur twinkle-1.2/configure.in twinkle-1.2.new/configure.in
--- twinkle-1.2/configure.in	2008-03-15 22:37:24.000000000 +0100
+++ twinkle-1.2.new/configure.in	2008-03-15 22:37:25.000000000 +0100
@@ -188,22 +188,33 @@
 
 # This check does not work on all platforms
 # Check if libgsm is available
-# AC_CHECK_LIB(gsm, sf_open, [
-#	AC_CHECK_HEADER(gsm.h, [],
-#		[AC_MSG_ERROR([gsm header files missing (gsm.h)])])
-#	AC_DEFINE(HAVE_GSM, 1, [Define to 1 if you have the <libgsm> library.])
-#	GSM_LIBS="-lgsm"
-#	echo "LIBS += -lgsm" >> $QT_INCL_PRO
-#	have_gsm="yes" ], [
-#	have_gsm="no"
-#	GSM_LIBS="\$(top_builddir)/src/audio/gsm/libgsm.a"
-#	echo "LIBS += ../audio/gsm/libgsm.a" >> $QT_INCL_PRO ])
-have_gsm="no"
-GSM_LIBS="\$(top_builddir)/src/audio/gsm/libgsm.a"
-echo "LIBS += ../audio/gsm/libgsm.a" >> $QT_INCL_PRO
+AC_CHECK_LIB(gsm, sf_open, [
+	AC_CHECK_HEADER(gsm.h, [],
+		[AC_MSG_ERROR([gsm header files missing (gsm.h)])])
+	AC_DEFINE(HAVE_GSM, 1, [Define to 1 if you have the <libgsm> library.])
+	GSM_LIBS="-lgsm"
+	echo "LIBS += -lgsm" >> $QT_INCL_PRO
+	have_gsm="yes" ], [
+	have_gsm="no"
+	GSM_LIBS="\$(top_builddir)/src/audio/gsm/libgsm.a"
+	echo "LIBS += ../audio/gsm/libgsm.a" >> $QT_INCL_PRO ])
+#have_gsm="no"
+#GSM_LIBS="\$(top_builddir)/src/audio/gsm/libgsm.a"
+#echo "LIBS += ../audio/gsm/libgsm.a" >> $QT_INCL_PRO
 
 AC_SUBST(GSM_LIBS)
 
+# Check if libgsm is available
+AC_CHECK_LIB(gsm, sf_open, [
+	AC_CHECK_HEADER(gsm.h, [],
+		[AC_MSG_ERROR([gsm header files missing (gsm.h)])])
+	AC_DEFINE(HAVE_GSM, 1, [Define to 1 if you have the <libgsm> library.])
+	LIBS="-lgsm $LIBS"
+	echo "LIBS += -lgsm" >> $QT_INCL_PRO
+	have_gsm="yes" ], [
+	have_gsm="no"
+	echo "$(top_builddir)/src/audio/gsm/libgsm.a" >> $QT_INCL_PRO ])
+
 # Check if ALSA is available
 AC_CHECK_LIB(asound, main, [
 	AC_CHECK_HEADER(alsa/asoundlib.h, [],
@@ -319,3 +330,4 @@
 AC_MSG_RESULT([Speex:       $have_speex])
 AC_MSG_RESULT([iLBC:        $have_ilbc])
 AC_MSG_RESULT([ZRTP:        $have_zrtp])
+AC_MSG_RESULT([Libgsm dynamic link:	$have_gsm])
diff -Nur twinkle-1.2/src/twinkle_config.h.in twinkle-1.2.new/src/twinkle_config.h.in
--- twinkle-1.2/src/twinkle_config.h.in	2008-03-09 09:33:02.000000000 +0100
+++ twinkle-1.2.new/src/twinkle_config.h.in	2008-03-15 22:37:25.000000000 +0100
@@ -16,6 +16,9 @@
    don't. */
 #undef HAVE_DECL_STRERROR_R
 
+/* Define to 1 if you have the <libgsm> library. */
+#undef HAVE_GSM
+
 /* Define to 1 if you have the <ilbc> library. */
 #undef HAVE_ILBC
 
